pkg/snapstore/utils.go

```diff
@@ -179,36 +178,36 @@ func adaptPrefix(snap *brtypes.Snapshot, snapstorePrefix string) string {
 	return snapstorePrefix
 }
 
-// GetSnapstoreSecretHash returns the hash of object store secrets hash
-func GetSnapstoreSecretHash(config *brtypes.SnapstoreConfig) (string, error) {
+func GetSnapstoreSecretModTime(config *brtypes.SnapstoreConfig) (time.Time, error) {
 	switch config.Provider {
 	case brtypes.SnapstoreProviderLocal:
-		return "", nil
+		return time.Time{}, nil
 	case brtypes.SnapstoreProviderS3:
-		return S3SnapStoreHash(config)
+		return S3SnapStoreModTime(config)
 	case brtypes.SnapstoreProviderABS:
-		return ABSSnapStoreHash(config)
+		return ABSSnapStoreModTime(config)
 	case brtypes.SnapstoreProviderGCS:
-		return GCSSnapStoreHash(config)
+		return GCSSnapStoreModTime(config)
 	case brtypes.SnapstoreProviderSwift:
-		return SwiftSnapStoreHash(config)
+		return SwiftSnapStoreModTime(config)
 	case brtypes.SnapstoreProviderOSS:
-		return OSSSnapStoreHash(config)
+		return OSSSnapStoreModTime(config)
 	case brtypes.SnapstoreProviderOCS:
-		return OCSSnapStoreHash(config)
+		return OCSSnapStoreModTime(config)
 	default:
-		return "", nil
+		return time.Time{}, nil
 	}
 }
 
-func getHash(data interface{}) string {
-	switch dat := data.(type) {
-	case string:
-		sha := sha256.Sum256([]byte(dat))
-		return fmt.Sprintf("%x", sha)
-	case []byte:
-		sha := sha256.Sum256(dat)
-		return fmt.Sprintf("%x", sha)
+// readModTime returns the latest modification time for an access credential
+func readModTime(fileInfo os.FileInfo) (time.Time, error) {
+	// reading credentials also checks for filesize, if 0 -> err
+	fileSize := fileInfo.Size()
+	if fileSize == 0 {
+		return time.Time{}, fmt.Errorf("credential file size zero")
 	}
-	return ""
+
+	fileModTime := fileInfo.ModTime()
+
+	return fileModTime, nil
```

Comment:
There are a few minor improvements and additions you could consider for clarity:

   - The error message `"credential file size zero"` is clear but could be more descriptive. Consider including the file name in the error message for easier debugging. Since `os.FileInfo` doesn't directly provide the file name, you might need to pass it as an additional argument to the function.

   - Update the function comment to reflect the complete functionality, including the check for file size. This helps in better understanding the purpose and functionality of the function.

   - Since you're directly returning `fileInfo.ModTime()`, you can return it without assigning it to `fileModTime`.


```go
// readModTime returns the latest modification time for an access credential file.
// It checks the file size and returns an error if the file size is zero.
func readModTime(fileInfo os.FileInfo, fileName string) (time.Time, error) {
    if fileInfo.Size() == 0 {
        return time.Time{}, fmt.Errorf("credential file '%s' has size zero", fileName)
    }

    return fileInfo.ModTime(), nil
}
```

Reply:
* `os.FileInfo` provides a `Name()` method which can directly be used to fetch the name of the file being read, hence the argument list need not be changed.
* Updated the function comment to reflect the complete functionality.
* Directly returning `fileInfo.ModTime()` now.